src/de/willuhn/jameica/fibu/util/BuchungUtil.java

```diff
@@ -116,4 +119,96 @@ public Object extract(ResultSet rs) throws RemoteException, SQLException
     
     return result;
   }
+
+  public static Map<String, Boolean> isSplit(Geschaeftsjahr jahr, Date von, Date bis) throws RemoteException{
+	  /*if (jahr == null)
+	      throw new RemoteException(i18n.tr("Kein Gesch�ftsjahr angegeben"));
+
+	    final Map<Integer,Boolean> result = new HashMap<>();
+
+	    DBIterator hauptbuchungen = jahr.getHauptBuchungen(von, bis);
+	    while (hauptbuchungen.hasNext()) {
+	    	Buchung b = (Buchung)hauptbuchungen.next();
+	    	Object o = b.getAttribute("split_id");
+	    	Integer i = (Integer)o;
+	    	//final Integer id = (Integer)(((Buchung)hauptbuchungen.next()).getAttribute("split_id"));
+	    	if(o == null)
+	    		continue;
+    		// Checken, ob wir schon einen Betrag haben
+	    	//final String s = (String)h;
+	    	//BaseBuchung bb = (BaseBuchung)o;
+            final Boolean split = result.get(i);
+            if (split == null)
+            	result.put(i,true);
+	    }
+	    return result;*/
+	    
+	    if (jahr == null)
+	        throw new RemoteException(i18n.tr("Kein Gesch�ftsjahr angegeben"));
+
+	      if (von != null && !jahr.check(von))
+	        throw new RemoteException(i18n.tr("Das Start-Datum {0} befindet sich ausserhalb des angegebenen Gesch�ftsjahres", Settings.DATEFORMAT.format(von)));
+
+	      if (bis != null && !jahr.check(bis))
+	        throw new RemoteException(i18n.tr("Das End-Datum {0} befindet sich ausserhalb des angegebenen Gesch�ftsjahres", Settings.DATEFORMAT.format(bis)));
+
+	      Date start = null;
+	      if (von != null)
+	        start = DateUtil.startOfDay(von);
+
+	      Date end = null;
+	      if (bis != null)
+	        end = DateUtil.endOfDay(bis);
+
+	      final Map<String,Boolean> result = new HashMap<>();
+
+	      final List params = new ArrayList();
+	      params.add(jahr.getID());
+	      
+	      try
+	      {
+	        DBService service = (DBService) Application.getServiceFactory().lookup(Fibu.class,"database");
+
+	        String sql = "SELECT id,(SELECT count(id) FROM buchung WHERE split_id = b.id) FROM buchung b WHERE geschaeftsjahr_id = ?";
```

Comment:
Wenn ich das richtig verstehe, geht es doch nur darum eine Map zurückzuliefern, die alle Splitbuchungen enthält. Alle Buchungen, die nicht in der Liste enthalten sind, sind damit ja implizit entweder die Hauptbuchungen oder ungesplittete Buchungen. In dem Fall würde ich den Ergebnis-Typ auf ein HashSet umstellen, welches nur noch die IDs der Splitbuchungen enthält. Das spart dann auch das Sub-Select.

Also etwa so: "SELECT id FROM buchung WHERE split_id IS NOT NULL".

Und dann einfach ein Set mit diesen IDs zurückliefern.